plugin.py

```diff
@@ -86,10 +86,28 @@ def install_or_update(cls) -> None:
     def markdown_language_id_to_st_syntax_map(cls) -> Optional["MarkdownLangMap"]:
         return {"python": (("python", "py"), ("LSP-pyright/syntaxes/pyright",))}
 
+    def on_server_response_async(self, method: str, response: Response) -> None:
+        if method == "textDocument/hover" and isinstance(response.result, dict):
+            contents = response.result.get("contents")
+            if not isinstance(contents, dict) or not contents.get("kind") == "markdown":
+                return
+            response.result["contents"]["value"] = self.patch_markdown_content(contents["value"])
+
     # -------------- #
     # custom methods #
     # -------------- #
 
+    def patch_markdown_content(self, content: str) -> str:
+        # Add another linebreak before horizontal rule following fenced code block
+        content = re.sub("```\n---", "```\n\n---", content)
```

Comment:
Given that this could get changed in pyright at some point (and honestly I would prefer trying to change it there first), we should be as defensive as possible.

Should we skip the replace if there are already 2 newlines in the content?

Also, shouldn't that replace all instance instead just the first one?

Reply:
> Given that this could get changed in pyright at some point (and honestly I would prefer trying to change it there first), we should be as defensive as possible.

Is this comment meant for the line 102 here, or in general? If it is for the line 102 shown in this diff, then I assume there won't be any change on the pyright side. It is already valid syntax (according to CommonMark specs), it just happens that mdpopups cannot render this correctly. That's why this workaround is needed.

> Should we skip the replace if there are already 2 newlines in the content?

There is no replacement happening if there are already 2 or more newlines?

> Also, shouldn't that replace all instances instead just the first one?

Not sure if I understand correctly, but `re.sub` *does* replace all instances and not just the first one.

Reply:
Yes, I meant line 102.

> There is no replacement happening if there are already 2 or more newlines?
> Not sure if I understand correctly, but `re.sub` _does_ replace all instances and not just the first one.

right, my bad.